File: C/palindrome.c
// palindrome.c
#include <stdio.h>
#include <string.h>
#include <ctype.h>

int isPalindrome(char str[]) {
    int l = 0;
    int h = strlen(str) - 1;
    
    while (h > l) {
        // Convert to lowercase for case-insensitive comparison
        if (tolower(str[l]) != tolower(str[h])) {
            return 0;
        }
        l++;
        h--;
    }
    return 1;
}

int main() {
    char str[] = "radar";
    char str2[] = "Madam";
    
    if (isPalindrome(str)) {
        printf("'%s' is a palindrome\n", str);
    } else {
        printf("'%s' is not a palindrome\n", str);
    }
    
    if (isPalindrome(str2)) {
        printf("'%s' is a palindrome\n", str2);
    } else {
        printf("'%s' is not a palindrome\n", str2);
    }
    
    return 0;
}

File: Python/palindrome.py
# palindrome.py

def is_palindrome(s):
    """Check if a string is a palindrome."""
    s = s.lower().replace(" ", "")  # Convert to lowercase and remove spaces
    return s == s[::-1]

if __name__ == "__main__":
    test_string = "radar"
    result = is_palindrome(test_string)
    print(f"'{test_string}' is {'a palindrome' if result else 'not a palindrome'}")
